active-rfcs/0000-function-api.md

```diff
@@ -546,31 +548,71 @@ To get proper type inference in TypeScript, we do need to wrap a component defin
 import { createComponent } from 'vue'
 
 const MyComponent = createComponent({
+  // props declarations are used to infer prop types
   props: {
     msg: String
   },
   setup(props) {
-    watch(() => props.msg, msg => { /* ... */ })
+    props.msg // string | undefined
+
+    // bindings returned from setup() can be used for type inference
+    // in templates
+    const count = value(0)
     return {
-      count: value(0)
+      count
     }
+  }
+})
+```
+
+`createComponent` is conceptually similar to 2.x's `Vue.extend`, but it is a no-op and only needed for typing purposes. The returned component is the object itself, but typed in a way that would provide type information for Vetur and TSX. If you are using Single-File Components, Vetur can implicitly add the wrapper function for you.
+
+If you are using render functions / TSX, returning a render function inside `setup()` provides proper type support (again, no manual type hints needed):
+
+``` ts
+import { createComponent, createElement as h } from 'vue'
+
+const MyComponent = createComponent({
+  props: {
+    msg: String
   },
-  render({ state, props }) {
-    // state typing inferred from value returned by setup()
-    console.log(state.count)
-    // props typing inferred from `props` declaration
-    console.log(props.msg)
-
-    // `this` exposes both state and props
-    console.log(this.count)
-    console.log(this.msg)
+  setup(props) {
+    const count = value(0)
+    return () => h('div', [
+      h('p', `msg is ${props.msg}`),
+      h('p', `count is ${count.value}`)
+    ])
+  }
+})
+```
+
+### TypeScript-only Props Typing
+
+In 3.0, the `props` declaration is optional. If you don't want runtime props validation, you can omit `props` declaration and declare your expected prop types directly in TypeScript:
+
+``` ts
+import { createComponent, createElement as h } from 'vue'
+
+interface Props {
+  msg: string
+}
+
+const MyComponent = createComponent({
```

Comment:
is `createComponent` still needed in this case? there is no type inference to be done I suppose?